test/unit/StakingPool/extendDeposit.js

```diff
@@ -0,0 +1,506 @@
+
+    await stakingPool
+      .connect(coverSigner)
+      .initialize(manager.address, false, initialPoolFee, maxPoolFee, productInitializationParams, poolId);
+
+    const { firstActiveTrancheId } = await getTranches();
+
+    {
+      const totalSupply = await stakingPool.totalSupply();
+      expect(totalSupply).to.equal(1);
+    }
+
+    await stakingPool.connect(user).depositTo([
+      {
+        amount,
+        trancheId: firstActiveTrancheId,
+        tokenId: 0,
+        destination: AddressZero,
+      },
+    ]);
+  });
+
+  it('reverts if token id is 0', async function () {
+    const { stakingPool } = this;
+    const {
+      members: [user],
+    } = this.accounts;
+
+    const { firstActiveTrancheId, maxTranche } = await getTranches();
+
+    await expect(stakingPool.connect(user).extendDeposit(0, firstActiveTrancheId, maxTranche, 0)).to.be.revertedWith(
+      'StakingPool: Invalid token id',
+    );
+  });
+
+  it('reverts if new tranche ends before the initial tranche', async function () {
+    const { stakingPool } = this;
+    const {
+      members: [user],
+    } = this.accounts;
+
+    const { depositNftId } = depositToFixture;
+
+    const { firstActiveTrancheId, maxTranche } = await getTranches();
+
+    await expect(
+      stakingPool.connect(user).extendDeposit(depositNftId, maxTranche, firstActiveTrancheId, 0),
+    ).to.be.revertedWith('StakingPool: The chosen tranche cannot end before the initial one');
+  });
+
+  it('reverts if new tranche is not yet available', async function () {
+    const { stakingPool } = this;
+    const {
+      members: [user],
+    } = this.accounts;
+
+    const { depositNftId } = depositToFixture;
+
+    const { firstActiveTrancheId, maxTranche } = await getTranches();
+
+    await expect(
+      stakingPool.connect(user).extendDeposit(depositNftId, firstActiveTrancheId, maxTranche + 1, 0),
+    ).to.be.revertedWith('StakingPool: The tranche is not yet available');
+  });
+
+  it('reverts if the new tranche already expired', async function () {
+    const { stakingPool } = this;
+    const {
+      members: [user],
+    } = this.accounts;
+
+    const { depositNftId } = depositToFixture;
+
+    const { firstActiveTrancheId } = await getTranches();
+
+    await increaseTime(TRANCHE_DURATION * 2);
+
+    await expect(
+      stakingPool.connect(user).extendDeposit(depositNftId, firstActiveTrancheId, firstActiveTrancheId + 1, 0),
+    ).to.be.revertedWith('StakingPool: The tranche has already expired');
+  });
+
+  it('any user can extend the deposit for a token id', async function () {
+    const { stakingPool } = this;
+    const {
+      nonMembers: [notNFTOwnerNorApproved],
+    } = this.accounts;
+
+    const { depositNftId } = depositToFixture;
+
+    const { firstActiveTrancheId } = await getTranches();
+
+    await stakingPool
+      .connect(notNFTOwnerNorApproved)
+      .extendDeposit(depositNftId, firstActiveTrancheId, firstActiveTrancheId + 1, 0);
+  });
```

Comment:
This is the current behavior, any user can call `extendDeposit` for any tokenId.  Unless I'm missing something, a possible attack I can think of could be that anyone can extend a staker deposit to future tranches before the current one expires, preventing the staker from withdrawing as desposits can only be withdrawn when the tranche expires. Shouldn't we check that the caller is OwnerOrApproved of the tokenId?